src/lcd.c

```diff
@@ -0,0 +1,228 @@
+#include "lcd.h"
+#include "gpio.h"
+
+#include <stdio.h>
+#include <string.h>
+#include <stdint.h>
+
+typedef struct  LCD_handler_s
+{
+    uint8_t rs_pin; // LOW: command.  HIGH: character.
+    uint8_t rw_pin; // LOW: write to LCD.  HIGH: read from LCD.
+    uint8_t enable_pin; // activated by a HIGH pulse.
+    uint8_t data_pins[8];
+
+    uint8_t displayfunction;
+    uint8_t displaycontrol;
+    uint8_t displaymode;
+
+    uint8_t initialized;
+
+    uint8_t numlines;
+    uint8_t row_offsets[4];
+}               LCD_handler;
+
+static LCD_handler handler;
```

Comment:
This module is rather inflexible:
- you can only have 1 LCD
- you cannot opt out. even if you never initialize the LCD module, the global variable is still allocated
- there is only one possible configuration for the LCD pins

Can you refactor the module, so that init function takes the pin configuration and returns an instance of LCD, and the rest of the functions takes the LCD instance as parameter.